### ISSUE TITLE
Container does not respect safe are bottom inset on iPhone X

### ISSUE THREAD

**Issue by jakubheglas** [kevin0571/STPopup#102]
When popup is presented as bottomSheet, the content is pushed lower than it should be on iPhone X. 

On gif attached below:
• big green button is properly pinned to the bottom edge of safe area
• container view's background is tinted red to indicate wrong offset as I pull it up

![bottom inset bug](https://user-images.githubusercontent.com/1138048/31288279-83805b66-aac4-11e7-8145-1aafe04ab0ff.gif)

**Comment by jakubheglas** [comment]
### Suggestion / food for thought

iPhone X's form factor obviously present entirely new design challenge to STPopup (and edge-to-edge designed elements generally). 

It would be amazing to have a style called .bottomFormSheet, that would effectively be a .formSheet, but pinned down to the bottom edge of safe area with a configurable insets.

The result could pretty much look like this from Apple.

![airpodscase-pf-open_airpods-pf_iphone7-jetblk-pf_pr-print-1](https://user-images.githubusercontent.com/1138048/31288392-0bd88146-aac5-11e7-97c4-6df0ff76dbc0.jpeg)

**Comment by kevin0571** [comment]
@jakubheglas it's actually not wrong offset. The extra height is intended for bottom sheet style popup when you drag the bottom sheet up. See https://github.com/kevin0571/STPopup/blob/master/STPopup/STPopupController.m#L526
But you are right, the current implementation for bottom sheet is not perfect for iPhone X. Will see how to fix this without introducing a new style.

**Comment by jakubheglas** [comment]
Oh, I absolutely understand the increased space because of the intentional drag! 

Let me rephrase: In the gif, the height of area which is represented by red color, should be added to `STPopupBottomSheetExtraHeight`. I assume `safeAreaInsets.bottom` should help there.

**Comment by abdallahnh** [comment]
Hi,
I'm facing same issue on iPhone x, does anyone solve it?

**Comment by danchokobo** [comment]
@kevin0571 did you resolve this issue?

**Comment by kevin0571** [comment]
Fixed in 1.8.4
